streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,287 @@
+        for (String key : topicProperties.stringPropertyNames()) {
+            topicConfig.put(key, topicProperties.getProperty(key));
+        }
+        topicDetails = new CreateTopicsRequest.TopicDetails(numPartitions, (short) replicationFactor, topicConfig);
+        Map<String, CreateTopicsRequest.TopicDetails> topics = new HashMap<>();
+        topics.put(internalTopicConfig.name(), topicDetails);
+
+        CreateTopicsRequest createTopicsRequest = new CreateTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        RequestCompletionHandler callback = new RequestCompletionHandler() {
+            public void onComplete(ClientResponse response) {
+                // Do nothing!
+            }
+        };
+        ClientResponse clientResponse = sendRequest(createTopicsRequest.toStruct(), ApiKeys.CREATE_TOPICS, callback);
+        CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(internalTopicConfig.name()).code() > 0) {
+            throw new StreamsException("Could not create topic: " + internalTopicConfig.name());
+        }
+    }
+
+    /**
+     * Delete a topic.
+     *
+     * @param topic
+     */
+    public void deleteTopic(String topic) {
+        log.debug("Deleting topic {} from ZK in partition assignor.", topic);
+        Set<String> topics = new HashSet();
+        topics.add(topic);
+
+        DeleteTopicsRequest deleteTopicsRequest = new DeleteTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        RequestCompletionHandler callback = new RequestCompletionHandler() {
+            public void onComplete(ClientResponse response) {
+                // Do nothing!
+            }
+        };
+        ClientResponse clientResponse = sendRequest(deleteTopicsRequest.toStruct(), ApiKeys.DELETE_TOPICS, callback);
+        CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(topic).code() > 0) {
+            throw new StreamsException("Could not delete topic: " + topic);
+        }
+
+    }
+
+    /**
+     * Send a request to kafka broker of this client. Polls the request for a given number of iterations to receive the response.
+     *
+     * @param request
+     * @param apiKeys
+     * @param callback
+     */
+    public ClientResponse sendRequest(Struct request, ApiKeys apiKeys, RequestCompletionHandler callback) {
+
+        String brokerId = Integer.toString(brokerNode.id());
+
+        RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(apiKeys),
+                request);
+
+        ClientRequest clientRequest = new ClientRequest(new SystemTime().milliseconds(), true, send, callback);
+        SystemTime systemTime = new SystemTime();
+        int iterationCount = 0;
+        while (iterationCount < maxIterations) {
```

Comment:
perhaps maxIterations should be a timeout instead? Also i think it reads cleaner like so:
```
final long timeout = systemTime.milliseconds + MAX_WAIT_TIME;
while (!kafka.ready(...)  && systemTime.milliseconds() < timeout) {
   kafkaClient.poll(...)
}
if (!kafkaClient.ready(...) {
    throw new StreamsException(...)
} 
kafkaClient.send(...)
```